src/BitBucketOnPrem.ts

```diff
@@ -135,12 +195,132 @@ export const getFileTreeFromBitbucket =
         return files;
     };
 
+export const cacheFileTree = async ({url, accessToken, projectKey, repoName, commit}: BitbucketOnPrem): Promise<boolean> => {
+    // If a repo is currently being cached, don't cache another one (log a warning and return false)
+    if (repoCurrentlyBeingCached) {
+        logger.warn("Cannot cache two repos at the same time");
+        return false;
+    }
+    try {
+        // This allows to query explorook about the repo currently being cached
+        repoCurrentlyBeingCached = {projectKey, repoName, commit};
+        const fileTreeUrl = UrlAssembler(url).template("/rest/api/1.0/projects/:projectKey/repos/:repoName/files").param({
```

Comment:
can you move the template string to a const ?

Reply:
Done